iceoryx2-userland/record-and-replay/src/testing.rs

```diff
@@ -0,0 +1,23 @@
+// Copyright (c) 2025 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache Software License 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0, or the MIT license
+// which is available at https://opensource.org/licenses/MIT.
+//
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+use crate::{
+    record::RawRecord,
+    recorder::{Recorder, RecorderWriteError},
+};
+
+pub fn recorder_write_unchecked(
```

Comment:
The `_unchecked` screams for `unsafe`. Can you describe what this function does and why it is not unsafe?